src/Workspaces/CSharp/Portable/Workspace/LanguageServices/CSharpSyntaxTreeFactoryService.cs

```diff
@@ -43,13 +41,13 @@ public override ParseOptions GetDefaultParseOptions()
             public override ParseOptions GetDefaultParseOptionsWithLatestLanguageVersion()
                 => _parseOptionWithLatestLanguageVersion;
 
-            public override SyntaxTree CreateSyntaxTree(string filePath, ParseOptions options, Encoding encoding, SyntaxNode root)
+            public override SyntaxTree CreateSyntaxTree(string? filePath, ParseOptions options, Encoding? encoding, SyntaxNode root)
             {
                 options ??= GetDefaultParseOptions();
                 return CSharpSyntaxTree.Create((CSharpSyntaxNode)root, (CSharpParseOptions)options, filePath, encoding);
```

Comment:
Note that right now this doesn't build because CSharpSyntaxTree.Create() is annotated as taking non-null, but it does internally convert the null back to an empty string. I started this annotation just to see if we had other issues or if that was it.

Reply:
Opened https://github.com/dotnet/roslyn/pull/59005 to update this.